rbac/management/workspace/model.py

```diff
@@ -81,6 +82,8 @@ def clean(self):
                 raise ValidationError({"workspace": f"{self.type} workspaces must have a parent workspace."})
         elif self.type == self.Types.DEFAULT and self.parent.type != self.Types.ROOT:
             raise ValidationError({"default_parent": "Default workspace must have a root parent."})
+        elif self.id == self.parent_id:
+            raise serializers.ValidationError({"parent_id": ("The parent_id and id values must not be the same.")})
```

Comment:
This moves the validation to where it should exist, but does not update the error raised. We'll need to refactor our exception handling to account for model validation errors. This preserves existing behavior, and we should follow-up on refactoring v2 exception handling and problem+json responses.